src/contexts/AssetContext/Provider.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { IAsset, IGetAccountAssetsResult, IGetAssetResult } from "types/NXTAPI";
+import useAccount from "hooks/useAccount";
+import useAPI from "hooks/useAPI";
+import useBlocks from "hooks/useBlocks";
+import Context from "./Context";
+
+const AssetProvider: React.FC = ({ children }) => {
+  const [heldAssets, setHeldAssets] = useState<Array<IAsset>>();
+  const { getAccountAssets, getAsset } = useAPI();
+  const { accountRs } = useAccount();
+  const { blockHeight } = useBlocks();
+
+  const fetchAccountAssets = useCallback(async () => {
+    let finalAssets: Array<IAsset> | undefined;
+
+    if (getAccountAssets === undefined || accountRs === undefined || getAsset === undefined) {
+      return;
+    }
+
+    const result: false | IGetAccountAssetsResult = await getAccountAssets(accountRs);
+
+    if (result) {
+      finalAssets = await processAssetResults(result.accountAssets, getAsset);
+    }
+
+    setHeldAssets(finalAssets);
+  }, [accountRs, getAccountAssets, getAsset]);
+
+  useEffect(() => {
```

Comment:
does the accountRs need to be checked here? i see it's checked in the fetchAccountAssets() function